wgpu-core/src/instance.rs

```diff
@@ -118,44 +119,30 @@ impl Instance {
         }
     }
 
-    pub fn from_hal<A: HalApi>(name: &str, raw_instance: A::Instance) -> Self {
-        #[cfg(vulkan)]
-        let mut vulkan_instance = None;
-        #[cfg(metal)]
-        let mut metal_instance = None;
-        #[cfg(dx12)]
-        let mut dx12_instance = None;
-        #[cfg(dx11)]
-        let mut dx11_instance = None;
-        #[cfg(gl)]
-        let mut gl_instance = None;
-        match A::VARIANT {
+    /// # Safety
+    ///
+    /// Refer to the creation of wgpu-hal Instance for every backend.
+    pub unsafe fn from_hal(name: &str, raw_instance: hal::RawInstance) -> Self {
```

Comment:
Does this need to be overly generic? Are you planning on passing anything but a Vulkan instance in here?
Perhaps this can be `from_vulkan` instead

Reply:
This code is already outdated.